pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -862,23 +862,23 @@ func (r *ReconcileClusterDeployment) getReleaseImage(cd *hivev1.ClusterDeploymen
 	return ""
 }
 
-func (r *ReconcileClusterDeployment) getClusterImageSet(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (*hivev1.ClusterImageSet, bool, error) {
+func (r *ReconcileClusterDeployment) getClusterImageSet(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (*hivev1.ClusterImageSet, error) {
 	if cd.Spec.ImageSet == nil || len(cd.Spec.ImageSet.Name) == 0 {
-		return nil, false, nil
+		return nil, nil
 	}
 	imageSet := &hivev1.ClusterImageSet{}
-	err := r.Get(context.TODO(), types.NamespacedName{Name: cd.Spec.ImageSet.Name}, imageSet)
-	switch {
-	case apierrors.IsNotFound(err):
-		cdLog.WithField("clusterimageset", cd.Spec.ImageSet.Name).Warning("clusterdeployment references non-existent clusterimageset")
-		modified, err := r.setImageSetNotFoundCondition(cd, false, cdLog)
-		return nil, modified, err
```

Comment:
The crux of the bug is here where `getClusterImageSet` would return `nil, false, nil` once the image set not found condition had been set. It should have been returning the IsNotFound error returned from the `Get` call.